### PR TITLE
Add Count*(ctx) methods to the stores used for telemetry

### PR DESCRIPTION
**Pull request by 0x656b694d**
## Description

Use `Count()` method to get the number of elements from stores instead of fetching all objects.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] ~Evaluated and added CHANGELOG entry if required~
- [ ] ~Determined and documented upgrade steps~
- [ ] ~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

## Testing Performed

Tested looking at the Segment debug messages, which came same as before.

### CODE REVIEW

**Comment by 0x656b694d** [comment]
Current dependencies on/for this PR:
* master
  * **PR #4214** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/4214" target="_blank">Graphite</a> 
    * **PR #4249** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/4249" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/4249?utm_source=stack-comment).

File: central/role/datastore/datastore_impl.go [hunk 1]
```diff
@@ -287,6 +303,14 @@ func (ds *dataStoreImpl) GetAllAccessScopes(ctx context.Context) ([]*storage.Sim
 	return scopes, nil
 }
 
+func (ds *dataStoreImpl) CountAccessScopes(ctx context.Context) (int, error) {
+	if ok, err := roleSAC.ReadAllowed(ctx); !ok || err != nil {
+		return 0, err
+	}
+
+	return ds.CountAccessScopes(ctx)
```
Comment (rhybrillou) [inline central/role/datastore/datastore_impl.go:311]: Shouldn't it be `ds.accessScopeStorage.Count(ctx)`?

File: central/role/datastore/datastore_impl.go [hunk 2]
```diff
@@ -161,6 +169,14 @@ func (ds *dataStoreImpl) GetAllPermissionSets(ctx context.Context) ([]*storage.P
 	return permissionSets, nil
 }
 
+func (ds *dataStoreImpl) CountPermissionSets(ctx context.Context) (int, error) {
+	if ok, err := roleSAC.ReadAllowed(ctx); !ok || err != nil {
+		return 0, err
+	}
+
+	return ds.CountPermissionSets(ctx)
```
Comment (rhybrillou) [inline central/role/datastore/datastore_impl.go:177]: Shouldn't it be `ds.permissionSetStorage.Count(ctx)` ?

**Review by rhybrillou — COMMENTED** [review]
I like the idea. A few points to address in the role-related datastore count functions and it will be ready to ship.

File: central/role/datastore/datastore_impl.go (hunk 1 above)
Comment (0x656b694d) [reply central/role/datastore/datastore_impl.go:311]: OMG

File: central/role/datastore/datastore_impl.go (hunk 2 above)
Comment (0x656b694d) [reply central/role/datastore/datastore_impl.go:177]: OMG returns

**Review by rhybrillou — APPROVED** [review]

**Pull request merged**